rfcs/0000-simple-override-strategy.md

```diff
@@ -13,118 +13,170 @@ require the evaluation of a function.
-      other_command ${bad}
+      other_command ${foo}
     '';
   };
 }
 ```
 
-Thus, with the previous rule, `buildPhase` [6] which evaluates to a string, is called with the
-attribute set which contains the derivations of `foo`, `bar` and `bad`.  This attribute set is then
-pass down to the dynamically scoped function call [8] which corresponds to the previous `buildPhase`
-[4], as `buildPhase` [7] is explicitly bound to the outer scope through the `super` [5] argument.
+This overriden package will evaluate the `buildPhase` [6] as to a string.  This string interpolates
+the evaluation of the `buildPhase` [8], which is inherited [7] from the out-side and corresponds to
+the evaluation of the previous `buildPhase`[4].
+
+The dynamic scope given to the `buildPhase` [6], and provided by the evaluation of the `buildDeps` [2]
+by the `drvBuilder` [1], is forwarded seamlessly to any dynamically scoped function under it, such as
+the `buildPhase` [8] evaluation, inherited from the overriden attribute set.
 
-At this stage, overriding the dependencies or the recipe of a package is performed with the update
-`//` operator.  Unfortunately the update operator is quite verbose, as we have to repeat all the
+Thus, the interpolation of `foo` and `bar` [5], correspond to the `outPath` of the derivation provided
+by the `drvBuilder` as the dynamic scope.
+
+At this stage, overriding the dependencies [2] or the recipe of a package [4] is performed with the update
+`//` operator.  This respectly replaces the need for the `override` and `overrideDerivation` functions, and
+by such any need for reflexivity.
+
+## Recursive Update
+
+Unfortunately the update operator is quite verbose, as we have to repeat all the
 names at each level of the attribute set.  To makes this easier to manipulate we would have to
-promote the `recursiveUpdate` function to give it some syntax as well.  Also, we would need to stop
-the recursiveUpdate to avoid recursive updates of `foo` in the `buildDeps` attribute.
+promote the `recursiveUpdate` function to give it some syntax as well.  Also, we need a way to stop
+the recursiveUpdate to avoid recursive updates in the attribute `foo` [9] of the `buildDeps` attribute.
+
+The update operator `//`, is only updating the top-level attribute set.  A recursive update operator
+need a way to *start zipping* (`/</`) attribute sets, and to *stop zipping* (`/>/:`) them to default to
+the update logic.
+
+Also, we need a way to refer to the original attribute set, such that we can alias the `buildPhase` [7],
+without repeating the package name.  In the following example, this is achieved with the stop operator,
+which alias the left-hand-side value of the update operator (`/>/@lhs:`).
+
+```nix
+self: super:
 
+{
+  fortytwo_1_0 = super.fortytwo_1_0 /</ {
+    buildDeps.foo = />/: self.foo_1_21; # [9]
+    buildPhase = />/@buildPhase: with : // { inherit buildPhase; }; ''
+      ${buildPhase}
+      other_command ${foo}
+    '';
+  };
+}
+```
```

Comment:
@zimbatm 
> I suppose the issue is that if the package is extended again, `buildDeps.foo` would be overridable. Is that the issue that you are trying to address?

Yes, this is part of the issues that are addressed by making `buildPhase` a function.

The other issue is the one described by @layus, which is that we want to stop overriding at `foo`, instead of within `foo`.  The difference being that if we stop within `foo` with might have extra information carried by the original expression into the result, thus leading a different evaluation.